### PR TITLE
Add Code Insights plugin to sample app and marketplace

### PR DESCRIPTION
**Pull request by lewtakm**
Hello,
This PR adds a Code Insights plugin to an app. Plugin provide new Code Insights tab for services and widgets (ReadMe, Languages, Releases) on the Overview page.

Right now plugin supports GitHub but we might plan to expand it to GitLab etc. 

#### :heavy_check_mark: Checklist

- [x] All tests are passing `yarn test`
- [x] Screenshots attached (for UI changes)
- [ ] Relevant documentation updated
- [x] Prettier run on changed files
- [x] Tests added for new functionality
- [ ] Regression tests added for bug fixes
- [ ] Added a changeset ([more info](https://github.com/spotify/backstage/blob/master/CONTRIBUTING.md#creating-changesets))

### CODE REVIEW

**Comment by dtuite** [comment]
It's probably worth having a discussion here about how much of this plugin is useful in the sample app.

1. The README widget was asked for in https://github.com/spotify/backstage/issues/2071 so that should probably go on the Overview tab.
2. The contributors panel should also go on the overview tab as it tells the story of the component.
3. Ditto for languages.
4. Releases won't be used by everyone so that is perhaps best as an optional widget.
5. Given the above we can probably retire the Code Insights tab for the sample app since most of the value will be on the Oveview tab.

Would like to hear input from the community here.

**Comment by Rugvip** [comment]
What's your intended scope of the plugin? It's very GitHub specific right now, but has a more generic name. I probably makes sense to deliberately install support for specific providers rather than trying to squeeze them all into one plugin, e.g. `github-insights`.

Regarding 2, 3, and 4, I think it might make sense to condense those into a single card? It's all really useful info imo, but could take less space. Regarding the releases I think it can simply be displayed if the project has any releases?, that becomes a bit more straightforward if it's all in one card, and you can also use GitHub's GraphQL API to fetch it all with a single request.

**Comment by dtuite** [comment]
@Rugvip It originally was called `github-insights`. Then @stefanalund suggested we change it to `code-insights` so that GitLab and Bitbucket can be added. 😄  

The idea is to provide access to "Insights" from SCM tools. [Here is GitLabs idea of inisghts](https://gitlab.com/gitlab-org/gitlab/insights/#/issues). [Here's is GitHubs](https://github.com/spotify/backstage/pulse). You'll see commonalities between them. Basically, "insights" try to answer questions like "How active is contribution to this repo?", "Who is contributing to this repo?" or "What types of contributions are being made?". They typically do this using visual features.

Regarding 2, 3 and 4, I agree it would be good as a single card. I believe that a condensed card option should augment the existing options rather than replacing them. If the user only wants to add the contributors card to the Overview tab then we should support them in that. If they want to add 2, 3 and 4 then perhaps they would prefer to use a condenced card which displays everything together.

**Comment by stefanalund** [comment]
@dtuite sorry for causing confusion here. I'm good with keeping the scope to GitHub and calling the plugin that for now. I still think the TAB should be named CODE INSIGHTS or just CODE though. Once you click into the plugin it can have a H1 saying GitHub Insights. TAB names should be idiomatic.

I think the "contributors" would actually fit very nicely into the About card on the Overview page. What do you think about moving it there?

**Comment by dtuite** [comment]
@stefanalund It's good to iterate on it and get it right.

**So here are the actions as I understand them:**

1. Rename the plugin to `backstage-plugin-github-insights`.
2. Ensure the tab is called "Code insights".
3. Add the Readme card to the Overview tab.
4. Add the Languages card to the Overview tab.
5. Add the releases card to the Overview tab but only show it if there are any releases.

**Separate GitHub issues for:**

1. Add contributors list to the About card on the Overview page. This must be a separate chunk of code since the code we we've already written is not present inside the spotify/backstage repo where the About card is. We are happy to do this work but it must be a separate PR.

**Work we won't do right now.**

1. If we're moving contributors into the About Card, it makes less sense to combine Contributors, Languages and Releases together so we will avoid doing this for the moment. Please create an Issue on https://github.com/RoadieHQ/backstage-plugin-code-insights if you would like to see some cards made which combine multiple insights together.



/cc @Rugvip @lewtakm

**Comment by stefanalund** [comment]
> 5. Add the releases card to the Overview tab but only show it if there are any releases.

@dtuite do releases make sense for other things than `Library` components? If not, then I think it should only be added there. The rest of the cards are generic enough to fit on the Overview pages for `Service`, `Website`, and `Library`.

**Comment by dtuite** [comment]
@stefanalund Releases are broader than that. You can release binaries for golang web services for example or docker containers for other things.

**Comment by stefanalund** [comment]
Very well. Let’s go forward with this plan then! Thanks 🙏🏼

File: microsite/data/plugins/code-insights.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+title: Code Insights
+author: roadie.io
+authorUrl: https://roadie.io
+category: Monitoring
+description: View Code Insights for your service in Backstage.
```
Comment (stefanalund) [inline microsite/data/plugins/code-insights.yaml:6]: Should be “for your components...” right?

**Comment by lewtakm** [comment]
@stefanalund : I have updated the PR according to what's @dtuite summarized: [here](https://github.com/spotify/backstage/pull/2833#issuecomment-707012517), I hope it will be okay now 👍

**Review by stefanalund — APPROVED** [review]
Excellent work, thanks a lot!

**Pull request merged**